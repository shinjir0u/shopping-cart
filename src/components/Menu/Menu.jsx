import { Link, useNavigate } from "react-router-dom";
import styles from "./Menu.module.css";
import Icon from "@mdi/react";
import { mdiCart } from "@mdi/js";

function Menu({ cartItemsCount = 0 }) {
  const navigate = useNavigate();

  return (
    <div className={styles.menu}>
      <div className={`${styles.menu__content} container`}>
        <img
          src="/src/assets/icons/logo.png"
          alt="logo"
          className={styles.logo}
        />
        <nav className={styles.nav}>
          <ul className={styles.nav__list}>
            <li className={styles.nav__list_item}>
              <Link className={styles.nav__link} to={"/"}>
                Home
              </Link>
            </li>
            <li className={styles.nav__list_item}>
              <Link className={styles.nav__link} to={"games"}>
                Games
              </Link>
            </li>
          </ul>
        </nav>
        <div
          onClick={() => navigate("cart")}
          className={styles.cart__container}
        >
          <Icon
            className={styles.cart}
            path={mdiCart}
            color={"rgb(213, 206, 163)"}
          ></Icon>
          <span className={styles.cart__items_counter}>{cartItemsCount}</span>
        </div>
      </div>
    </div>
  );
}

export default Menu;
