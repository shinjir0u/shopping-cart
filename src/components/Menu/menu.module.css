ul {
  padding: 0;
}

li {
  list-style-type: none;
}

a {
  text-decoration: none;
  color: black;
}

.menu {
  background-color: var(--dark-background-color);
}

.menu__content {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1em;
}

.logo {
  /* background-color: var(--beige-color);
  border-radius: 50%;
  padding: 0.5em; */
  width: 4.5em;
}

.nav__list_item {
  text-align: center;
}

.nav__link {
  font-size: 1.5rem;
  font-weight: bold;
  color: var(--white-color);
}

.cart__container {
    margin-top: 1em;
    position: relative;
    display: flex;
}

svg.cart {
    width: 3em;
}

.cart__items_counter {
    position: absolute;
    background-color: var(--white-color);
    padding: 0.2em 0.5em;
    border-radius: 50%;
    color: var(--brown-color);
    font-weight: bold;
    transform: translate(150%, -50%);
}
