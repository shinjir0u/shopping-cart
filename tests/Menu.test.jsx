import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Menu from "../src/components/Menu/Menu";
import { MemoryRouter } from "react-router-dom";

describe("Menu component exists", () => {
  it("Menu componenet renders correctly", () => {
    const result = render(
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    );
    expect(result).toMatchSnapshot();
  });
});
