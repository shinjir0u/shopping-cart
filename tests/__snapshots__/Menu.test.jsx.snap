// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu component exists > Menu componenet renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_menu_124232"
      >
        <div
          class="_menu__content_124232 container"
        >
          <img
            alt="logo"
            class="_logo_124232"
            src="/src/assets/icons/logo.png"
          />
          <nav
            class="_nav_124232"
          >
            <ul
              class="_nav__list_124232"
            >
              <li
                class="_nav__list_item_124232"
              >
                <a
                  class="_nav__link_124232"
                  data-discover="true"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="_nav__list_item_124232"
              >
                <a
                  class="_nav__link_124232"
                  data-discover="true"
                  href="/games"
                >
                  Games
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="_cart__container_124232"
          >
            <svg
              class="_cart_124232"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M17,18C15.89,18 15,18.89 15,20A2,2 0 0,0 17,22A2,2 0 0,0 19,20C19,18.89 18.1,18 17,18M1,2V4H3L6.6,11.59L5.24,14.04C5.09,14.32 5,14.65 5,15A2,2 0 0,0 7,17H19V15H7.42A0.25,0.25 0 0,1 7.17,14.75C7.17,14.7 7.18,14.66 7.2,14.63L8.1,13H15.55C16.3,13 16.96,12.58 17.3,11.97L20.88,5.5C20.95,5.34 21,5.17 21,5A1,1 0 0,0 20,4H5.21L4.27,2M7,18C5.89,18 5,18.89 5,20A2,2 0 0,0 7,22A2,2 0 0,0 9,20C9,18.89 8.1,18 7,18Z"
                style="fill: rgb(213, 206, 163);"
              />
            </svg>
            <span
              class="_cart__items_counter_124232"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_menu_124232"
    >
      <div
        class="_menu__content_124232 container"
      >
        <img
          alt="logo"
          class="_logo_124232"
          src="/src/assets/icons/logo.png"
        />
        <nav
          class="_nav_124232"
        >
          <ul
            class="_nav__list_124232"
          >
            <li
              class="_nav__list_item_124232"
            >
              <a
                class="_nav__link_124232"
                data-discover="true"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="_nav__list_item_124232"
            >
              <a
                class="_nav__link_124232"
                data-discover="true"
                href="/games"
              >
                Games
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="_cart__container_124232"
        >
          <svg
            class="_cart_124232"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M17,18C15.89,18 15,18.89 15,20A2,2 0 0,0 17,22A2,2 0 0,0 19,20C19,18.89 18.1,18 17,18M1,2V4H3L6.6,11.59L5.24,14.04C5.09,14.32 5,14.65 5,15A2,2 0 0,0 7,17H19V15H7.42A0.25,0.25 0 0,1 7.17,14.75C7.17,14.7 7.18,14.66 7.2,14.63L8.1,13H15.55C16.3,13 16.96,12.58 17.3,11.97L20.88,5.5C20.95,5.34 21,5.17 21,5A1,1 0 0,0 20,4H5.21L4.27,2M7,18C5.89,18 5,18.89 5,20A2,2 0 0,0 7,22A2,2 0 0,0 9,20C9,18.89 8.1,18 7,18Z"
              style="fill: rgb(213, 206, 163);"
            />
          </svg>
          <span
            class="_cart__items_counter_124232"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
